contents/tutorials/ruby-ab-tests.md

```diff
@@ -0,0 +1,278 @@
+
+```Gemfile
+gem "posthog-ruby"
+```
+
+Then `bundle install` to install the gem.
+
+Next, we create an initializer for our PostHog client. Create a new file in `config/initializers` called `posthog.rb`. Then, initialize PostHog in this file using your project API key and host:
+
+```ruby file=config/initializers/posthog.rb
+require 'posthog-ruby'
+
+POSTHOG_CLIENT = PostHog::Client.new({
+  api_key: '<ph_project_api_key>',
+  host: '<ph_instance_address>'
+})
+```
+
+Then, we fetch the feature flag in our controller `app/controllers/pages_controller.rb` and set the button text using the value from it:
+
+```ruby file=app/controllers/pages_controller.rb
+class PagesController < ApplicationController
+  def home
+    @button_text = 'No variant'
+    begin
+      distinct_id = 'placeholder-user-id'
+      enabled_variant = POSTHOG_CLIENT.get_feature_flag('my-cool-experiment', distinct_id)
+      if enabled_variant == 'control'
+        @button_text = 'Control Variant'
+      elsif enabled_variant == 'test'
+        @button_text = 'Test Variant'
+      end
+    rescue => e
+      @button_text = 'Error'
+    end
+  end
+end
+```
+
+Lastly, update our view to use the `@button_text` variable from the controller:
+
+```html file=app/views/pages/home.html.erb
+<script>
+  function handleClick() {
+    console.log(posthog)
+    posthog.capture('home_button_clicked');
+  }
+</script>
+
+<main>
+  <h1>Rails A/B Test</h1>
+  <button id="main-cta" onclick="handleClick()"><%= @button_text %></button>
+</main>
+```
+
+Now, when you refresh the page, the button text is already set when the page loads. 
+
+#### Setting the correct `distinctId`
+
+You may notice that we set `distinctId = 'placeholder-user-id'` in our flag call above. In production apps, to ensure you fetch the correct flag value for your user, `distinctId` should be set to their unique ID. 
+
+For logged-in users, you typically use their email as their `distinctId`. However, for logged-out users, you can use the `distinct_id` property from their PostHog cookie:
+
+
+```ruby file=app/controllers/pages_controller.rb
+class PagesController < ApplicationController
+  def home
+    @button_text = 'No variant'
+    
+    project_api_key = '<ph_project_api_key>'
+    cookie_key = "ph_#{project_api_key}_posthog"
+
+    if cookies[cookie_key]
+      begin
+        distinct_id = JSON.parse(cookies[cookie_key])['distinct_id']
+        enabled_variant = POSTHOG_CLIENT.get_feature_flag('my-cool-experiment', distinct_id)
+        if enabled_variant == 'control'
+          @button_text = 'Control variant'
+        elsif enabled_variant == 'test'
+          @button_text = 'Test variant'
+        end
+      rescue => e
+        @button_text = 'Error'
+      end
+    end
+  end
+end
+```
+
+In the scenario where your PostHog cookie is not available on the server side, you can fall back to the client side rendering approach of feature flags
```

Comment:
Could we add the code that checks `@button_text` on the client side and checks the flag if it doesn't exist? 
```suggestion
In the scenario where your PostHog cookie is not available on the server side, you can fall back to the client side evaluation of the feature flag.
```